api/swagger.yaml

```diff
@@ -0,0 +1,119 @@
+openapi: 3.0.2
+info:
+  title: Web Server
+  description: A simple web server for managing users
+  contact:
+    email: dmanor@redhat.com
+  version: "1.0"
+servers:
+- url: http://127.0.0.1:3000
+paths:
+  /api/users:
+    get:
+      summary: Retrieves all the users from the system
+      operationId: findUsers
+      responses:
+        "200":
+          description: Users retrieved successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+                x-content-type: application/json
+        "405":
+          description: A problem occured while retrieving users
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    post:
+      summary: Create a new user in the system
+      description: Create a new user in the system
+      operationId: addUser
+      requestBody:
+        description: Create a new user in the system
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/NewUser'
+        required: true
+      responses:
+        "200":
+          description: User added successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        "405":
```

Comment:
again, I'm not sure it really means that
``409`` (conflict) or just the general ``400`` code can deliver here better, since the user tried to create an existing record in the system